pkg/planner/core/rule_decorrelate.go

```diff
@@ -133,8 +133,65 @@ func (*DecorrelateSolver) aggDefaultValueMap(agg *logicalop.LogicalAggregation)
 	return defaultValueMap
 }
 
+// pruneRedundantApply: Removes the Apply operator if the parent SELECT clause does not filter any rows from the source.
+// Example: SELECT 1 FROM t1 AS tab WHERE 1 = 1 OR (EXISTS(SELECT 1 FROM t2 WHERE a2 = a1))
+// In this case, the subquery can be removed entirely since the WHERE clause always evaluates to True.
+// This results in a SELECT node with a True condition and an Apply operator as its child.
+// If this pattern is detected, we remove both the SELECT and Apply nodes, returning the left child of the Apply operator as the result.
+// For the example above, the result would be a table scan on t1.
+func pruneRedundantApply(p base.LogicalPlan) (base.LogicalPlan, bool) {
+	// Check if the current plan is a LogicalSelection
+	logicalSelection, ok := p.(*logicalop.LogicalSelection)
+	if !ok {
+		return nil, false
+	}
+
+	// Retrieve the child of LogicalSelection
+	selectSource := logicalSelection.Children()[0]
+
+	// Check if the child is a LogicalApply
+	apply, ok := selectSource.(*logicalop.LogicalApply)
```

Comment:
I'm afriad that this pattern is more tricky
```
tidb> explain select * from (select * from t where EXISTS(SELECT /*+ NO_DECORRELATE() */ 1  FROM t t2 WHERE t2.a = t.a)) s where 1 ;
+-----------------------------+--------------+-----------+---------------+-----------------------------------------------+
| id                          | estRows      | task      | access object | operator info                                 |
+-----------------------------+--------------+-----------+---------------+-----------------------------------------------+
| Apply_12                    | 10000.00     | root      |               | CARTESIAN semi join, left side:TableReader_14 |
| ├─TableReader_14(Build)     | 10000.00     | root      |               | data:TableFullScan_13                         |
| │ └─TableFullScan_13        | 10000.00     | cop[tikv] | table:t       | keep order:false, stats:pseudo                |
| └─TableReader_17(Probe)     | 100000.00    | root      |               | data:Selection_16                             |
|   └─Selection_16            | 100000.00    | cop[tikv] |               | eq(test.t.a, test.t.a)                        |
|     └─TableFullScan_15      | 100000000.00 | cop[tikv] | table:t2      | keep order:false, stats:pseudo                |
+-----------------------------+--------------+-----------+---------------+-----------------------------------------------+
6 rows in set (0.01 sec)
```
in this case:
selection<1> which is finally removed, has a apply as its child, while we couldn't just let apply inner child to directly be the root here? cause the apply source is not comes from outer query block's where clause's scalar Exists, at which we thought Exists is same level with Where 1 here.
/cc @winoros